radix-engine-common/src/iterators/overlaying_iterator.rs

```diff
@@ -0,0 +1,88 @@
+use crate::internal_prelude::*;
+use sbor::rust::cmp::*;
+use sbor::rust::iter::*;
+
+/// An iterator overlaying a "change on a value" (coming from the [`overlaying`] iterator) over a
+/// "base value" (coming from the [`underlying`] iterator).
+/// The one is matched to another by a `K` part (of the iterated tuple `(K, V)`), which both
+/// iterators are assumed to be ordered by.
+pub struct OverlayingIterator<U, O>
```

Comment:
What do you think of moving this to `utils`, as per Yulong's comment?

Reply:
Sounds good, moving to `utils`